app/models/container_image.rb

```diff
@@ -45,9 +47,10 @@ def operating_system=(value)
   end
 
   def docker_id
-    if image_ref.start_with?(DOCKER_IMAGE_PREFIX)
-      return image_ref[DOCKER_IMAGE_PREFIX.length..-1]
+    DOCKER_PREFIXES.each do |prefix|
+      return image_ref[prefix.length..-1] if image_ref.start_with?(prefix)
     end
```

Comment:
OK agreed to keep the validation function in scanning job, but separate validation logic from queuing & error handling